test/native-promises.tap.js

```diff
@@ -1480,6 +1481,31 @@ test('Promise.resolve().catch().then()', function (t) {
     process.removeAsyncListener(listenerState.listener);
 });
 
+test('continue from unwrapped promise', function(t) {
+  var listener = addListner();
+
+  listener.currentName = 'resolve';
+  resolvedBeforeWrap.then(function(val) {
+    t.equal(val, 123, 'should match resolved value')
+    listener.currentName = '2nd resolve';
+    return 456
```

Comment:
What I mean by that is: if we are not in an active context when a callback runs, fallback to the context that was active when then was called.  I think this is probably a bad idea though.  For now I say we move forward with it as is.